src/pykeen/nn/vision.py

```diff
@@ -0,0 +1,297 @@
+        return self.transforms(image)
+
+    # docstr-coverage: inherited
+    def __len__(self) -> int:  # noqa:D105
+        return len(self.images)
+
+
+class VisualRepresentation(Representation):
+    """Visual representations using a torchvision model."""
+
+    def __init__(
+        self,
+        images: Sequence,
+        encoder: Union[str, torch.nn.Module],
+        layer_name: str,
+        max_id: Optional[int] = None,
+        transforms: Optional[Sequence] = None,
+        encoder_kwargs: OptionalKwargs = None,
+        batch_size: int = 32,
+        trainable: bool = True,
+        **kwargs,
+    ):
+        """
+        Initialize the representations.
+
+        :param images:
+            the images, either as tensors, or paths to image files.
+        :param encoder:
+            the encoder to use. If given as a string, lookup in :module:`torchvision.models`
+        :param layer_name:
+            the model's layer name to use for extracting the features, cf.
+            :func:`torchvision.models.feature_extraction.create_feature_extractor`
+        :param max_id:
+            the number of representations. If given, it must match the number of images.
+        :param transforms:
+            transformations to apply to the images. Notice that stochastic transformations will result in
+            stochastic representations, too.
+        :param encoder_kwargs:
+            additional keyword-based parameters passed to encoder upon instantiation.
+        :param batch_size:
+            the batch size to use during encoding
+        :param trainable:
+            whether the encoder should be trainable
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`Representation.__init__`.
+            Should not include `max_id` or `shape`.
+
+        :raises ValueError:
+            if `max_id` is provided and does not match the number of images
+        """
+        _ensure_vision(self, models)
+        self.images = VisionDataset(images=images, transforms=transforms)
+
+        if isinstance(encoder, str):
+            cls = getattr(models, encoder)
+            encoder = cls(encoder_kwargs or {})
+
+        pool = functools.partial(torch.mean, dim=(-1, -2))
+
+        encoder = models.feature_extraction.create_feature_extractor(
+            model=encoder, return_nodes={layer_name: "feature"}
+        )
+
+        # infer shape
+        with torch.inference_mode():
+            encoder.eval()
+            shape = self._encode(images=self.images[0].unsqueeze(dim=0), encoder=encoder, pool=pool).shape[1:]
+
+        if max_id is not None and len(images) != max_id:
+            raise ValueError(
+                f"Inconsistent max_id={max_id} and len(images)={len(images)}. In case there are not images for all "
+                f"IDs, you may consider using BackfillRepresentation.",
+            )
+        super().__init__(max_id=max_id, shape=shape, **kwargs)
```

Comment:
`max_id` isn't guaranteed to be non-none here - if it is not given, can it be inferred from the length of the images?

Reply:
fixed by https://github.com/pykeen/pykeen/pull/965/commits/657e582042f106e7da2adbaeac24f47832805dea